scripts/test-clienthello-md5.py

```diff
@@ -0,0 +1,218 @@
+
+
+def natural_sort_keys(s, _nsre=re.compile('([0-9]+)')):
+    return [int(text) if text.isdigit() else text.lower()
+            for text in re.split(_nsre, s)]
+
+
+def help_msg():
+    print("Usage: <script-name> [-h hostname] [-p port] [[probe-name] ...]")
+    print(" -h hostname    name of the host to run the test against")
+    print("                localhost by default")
+    print(" -p port        port number to use for connection, 4433 by default")
+    print(" probe-name     if present, will run only the probes with given")
+    print("                names and not all of them, e.g \"sanity\"")
+    print(" -e probe-name  exclude the probe from the list of the ones run")
+    print("                may be specified multiple times")
+    print(" --workaround   workaround https://bugzilla.redhat.com/show_bug.cgi?id=1411238")
+    print(" --help         this message")
+
+
+def main():
+    """Check what signature algorithms server advertises"""
+    hostname = "localhost"
+    port = 4433
+    run_exclude = set()
+    sigalgs = [(HashAlgorithm.sha512, SignatureAlgorithm.rsa),
+               (HashAlgorithm.sha384, SignatureAlgorithm.rsa),
+               (HashAlgorithm.sha256, SignatureAlgorithm.rsa),
+               (HashAlgorithm.sha224, SignatureAlgorithm.rsa),
+               (HashAlgorithm.sha1, SignatureAlgorithm.rsa)]
+    cert = None
+    private_key = None
+    workaround = False
+
+    argv = sys.argv[1:]
+    opts, args = getopt.getopt(argv, "h:p:e:", ["help", "workaround"])
+    for opt, arg in opts:
+        if opt == '-h':
+            host = arg
+        elif opt == '-p':
+            port = int(arg)
+        elif opt == '--workaround':
+            workaround = True
+        elif opt == '-e':
+            run_exclude.add(arg)
+        elif opt == '--help':
+            help_msg()
+            sys.exit(0)
+        else:
+            raise ValueError("Unknown option: {0}".format(opt))
+
+    if args:
+        run_only = set(args)
+    else:
+        run_only = None
+
+    conversations = {}
+
+    # sanity check of connection
+    conversation = Connect(hostname, port)
+    node = conversation
+    ciphers = [CipherSuite.TLS_DHE_RSA_WITH_AES_256_CBC_SHA256,
+               CipherSuite.TLS_EMPTY_RENEGOTIATION_INFO_SCSV]
+    node = node.add_child(ClientHelloGenerator(ciphers))
+
+    node = node.add_child(ExpectServerHello(version=(3, 3)))
+    node = node.add_child(ExpectCertificate())
+    node = node.add_child(ExpectServerKeyExchange())
+    node = node.add_child(ExpectServerHelloDone())
+    node = node.add_child(ClientKeyExchangeGenerator())
+    node = node.add_child(ChangeCipherSpecGenerator())
+    node = node.add_child(FinishedGenerator())
+    node = node.add_child(ExpectChangeCipherSpec())
+    node = node.add_child(ExpectFinished())
+    node = node.add_child(ApplicationDataGenerator(b"GET / HTTP/1.0\n\n"))
+    node = node.add_child(ExpectApplicationData())
+    node = node.add_child(ExpectClose())
```

Comment:
clean connection close requires sending `connection_close` warning alert and expecting one in return, without this, the test case is essentially time-outing the connection on server side (applies to all test cases)